### DISCUSSION TITLE
0.68.2: failures running tests `Failed: DID NOT RAISE <class 'starlette.websockets.WebSocketDisconnect'>`

### DISCUSSION THREAD

**Discussion by sandrotosi** [fastapi/fastapi#6609] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
N/A here
```

### Description

I'm packaging fastapi/0.68.2 for Debian. During our build process, we also run the test suite, and 2 of the tests fail as:

```
=================================== FAILURES ===================================
________________________ test_websocket_no_credentials _________________________

    def test_websocket_no_credentials():
        with pytest.raises(WebSocketDisconnect):
>           client.websocket_connect("/items/foo/ws")
E           Failed: DID NOT RAISE <class 'starlette.websockets.WebSocketDisconnect'>

tests/test_tutorial/test_websockets/test_tutorial002.py:75: Failed
_________________________ test_websocket_invalid_data __________________________

    def test_websocket_invalid_data():
        with pytest.raises(WebSocketDisconnect):
>           client.websocket_connect("/items/foo/ws?q=bar&token=some-token")
E           Failed: DID NOT RAISE <class 'starlette.websockets.WebSocketDisconnect'>

tests/test_tutorial/test_websockets/test_tutorial002.py:80: Failed
=========================== short test summary info ============================
FAILED tests/test_tutorial/test_websockets/test_tutorial002.py::test_websocket_no_credentials
FAILED tests/test_tutorial/test_websockets/test_tutorial002.py::test_websocket_invalid_data
================= 2 failed, 990 passed, 3 deselected in 7.97s ==================
```

we're using starlette/0.16.0 which is the latest upstream release

### Operating System

Linux

### FastAPI Version

0.68.2

### Python Version

3.9.7

### Additional Context

This is running in a chroot without internet access (given it's a websocket test, it may or may not be relevant)

**Comment by ArcLightSlavik** [comment]
Few notes:

1) >we're using starlette/0.16.0 which is the latest upstream release

starlette 0.16.0 is for fastapi 0.70.0, I would either bump fastapi or lower starlette

2) >we also run the test suite

personally don't see any reason to run these tests

Ran both latest master and the 0.68.2 release in both online/offline mode and all tests passed
Seems like something with your configuration/debian shenanigans

**Comment by sandrotosi** [comment]
> Few notes:
> 
> 1. > we're using starlette/0.16.0 which is the latest upstream release
> 
> starlette 0.16.0 is for fastapi 0.70.0, I would either bump fastapi or lower starlette

how can one know that "starlette 0.16.0 is for fastapi 0.70.0"; i see 0.70.0 was released less that a day ago, i'll try to upgrade to it and report back

> 1. > we also run the test suite
> 
> personally don't see any reason to run these tests

i disagree: they provide a validation of the ability to run a module in a specific environment: Debian only provides a single version of each module, so it's important to verify they are all compatible; i think in this case, tests worked as expected and showed a possible issue that needed investigation
 
> Ran both latest master and the 0.68.2 release in both online/offline mode and all tests passed Seems like something with your configuration/debian shenanigans

It may very well be a Debian-specific issue, that's why i was asking for help identifying what it is

**Comment by ArcLightSlavik** [comment]
>how can one know that "starlette 0.16.0 is for fastapi 0.70.0"; i see 0.70.0 was released less that a day ago

either manually via `pip install fastapi` and then doing `pip freeze` or using some kind of package manager (poetry etc) that does the version calculation for you
